dartagnan/src/main/java/com/dat3m/dartagnan/program/ScopeHierarchy.java

```diff
@@ -40,27 +39,27 @@ public int getScopeId(String scope) {
         return scopeIds.getOrDefault(scope, -1);
     }
 
-    // For each scope S higher than flag, we check both events are in the same scope S
-    public boolean sameAtHigherScope(ScopeHierarchy thread, String flag) {
-        if (!this.getClass().equals(thread.getClass()) || !this.getScopes().contains(flag)) {
+    // For any scope higher than the given one, we check both threads have the same scope id.
```

Comment:
Maybe "// For any scope higher than the given one, we check both **ScopeHierarchy** have the same scope id."? Because the Thread class is not involved here.

Reply:
I think "thread" is a better explanation of what if going on at the high level. The scope hierarchy is an artifact of our implementation, but at the end of the day, what we really care is to know which is the scope id for a given thread.

Reply:
Okay, yes "thread" makes more sense at the high level.